### PR TITLE
Add RandomCropAndResize to ImageNet classification training script

### PR DESCRIPTION
**Pull request by ianstenbit**
@LukeWood @tanzhenyu 

Training results with this change are pending (although I've verified that the training script still runs correctly). I'll train an EfficientNet with this change and compare results to the previous version.

### CODE REVIEW

**Comment by tanzhenyu** [comment]
To clarify -- we only use this if it is not worse than existing ResNet50?

**Comment by ianstenbit** [comment]
> To clarify -- we only use this if it is not worse than existing ResNet50?

Yeah I figured I'd do a training run against a model for which we already have weights and verify that this performs better (I figured I'd do an EfficientNet since it's quick to train)

**Comment by tanzhenyu** [comment]
> > To clarify -- we only use this if it is not worse than existing ResNet50?
> 
> Yeah I figured I'd do a training run against a model for which we already have weights and verify that this performs better (I figured I'd do an EfficientNet since it's quick to train)

Given this has a broader impact, let's make sure it also runs well on ResNet?

**Comment by ianstenbit** [comment]
> Given this has a broader impact, let's make sure it also runs well on ResNet?

SGTM! I'll kick off a ResNet50V2 training run

**Comment by ianstenbit** [comment]
> > Given this has a broader impact, let's make sure it also runs well on ResNet?
> 
> SGTM! I'll kick off a ResNet50V2 training run

The training run isn't done yet, but it's currently outperforming our previous ResNet50V2 runs (72% top-1 accuracy after 78 epochs and it still hasn't Reduced Lr on Plateau yet)

So I think it's fair to say this is working well. I think we can merge the change, and once the ResNet run finishes I will upload our new best weights

**Comment by tanzhenyu** [comment]
> > > Given this has a broader impact, let's make sure it also runs well on ResNet?
> > 
> > 
> > SGTM! I'll kick off a ResNet50V2 training run
> 
> The training run isn't done yet, but it's currently outperforming our previous ResNet50V2 runs (72% top-1 accuracy after 78 epochs and it still hasn't Reduced Lr on Plateau yet)
> 
> So I think it's fair to say this is working well. I think we can merge the change, and once the ResNet run finishes I will upload our new best weights

Isn't the ResNet50V2 at 76% after 90 epochs?

**Comment by ianstenbit** [comment]
> Isn't the ResNet50V2 at 76% after 90 epochs?

Our best run was 75.5% after 168 epochs. It didn't pass 71% until after its first LR reduction

**Review by LukeWood — APPROVED** [review]

**Comment by tanzhenyu** [comment]
> > Isn't the ResNet50V2 at 76% after 90 epochs?
> 
> Our best run was 75.5% after 168 epochs. It didn't pass 71% until after its first LR reduction

Oh...that doesn't seem right at all, I'd assume it should pass 71% at 60 epochs

**Comment by ianstenbit** [comment]
> Oh...that doesn't seem right at all, I'd assume it should pass 71% at 60 epochs

In our previous run it got quite close to that: https://tensorboard.dev/experiment/TQ5r1EhXS4SDDagBD84rgA/#scalars
This is likely due to using a high LR with DecreaseLROnPlateau instead of a decaying LR.

Once I am able to test #908 we will include the warmed-up Cosine decay which may improve the speed of convergence (if not the actual final scores)

**Comment by ianstenbit** [comment]
The new ResNet50V2 training run with this change has now surpassed our current weights' validation accuracy by 0.5% (and the training is still running).

Any objections to my merging this @tanzhenyu ?

**Review by tanzhenyu — APPROVED** [review]
Thanks!
